src/cli/rustup_mode.rs

```diff
@@ -1586,22 +1586,38 @@ fn doc(cfg: &Cfg, m: &ArgMatches) -> Result<utils::ExitCode> {
     };
 
     let topical_path: PathBuf;
-
-    let doc_url = if let Some(topic) = m.get_one::<String>("topic") {
-        topical_path = topical_doc::local_path(&toolchain.doc_path("").unwrap(), topic)?;
-        topical_path.to_str().unwrap()
-    } else if let Some((_, _, path)) = DOCS_DATA.iter().find(|(name, _, _)| m.get_flag(name)) {
-        path
-    } else {
-        "index.html"
-    };
+    let cached_path: String;
```

Comment:
Why do we need to define this one here? Maybe we can put it into the `if` statement.

Reply:
This is the way to resolve &str borrowing issue. If we put this definition into if statement we will get error:
error[E0597]: `cached_path` does not live long enough
maybe there is a nicer workaround, but I have not found it yet